pkg/dataplane/util/ansible_execution.go

```diff
@@ -58,190 +56,189 @@ func AnsibleExecution(
 	var sshKeyMountPath string
 	var sshKeyMountSubPath string
 
+	const jobRestartPolicy string = "OnFailure"
+
 	ansibleEEMounts := storage.VolMounts{}
 
 	executionName, labels := GetAnsibleExecutionNameAndLabels(service, deployment.GetName(), nodeSet.GetName())
-	ansibleEE, err := GetAnsibleExecution(ctx, helper, deployment, labels)
+
+	existingAnsibleEE, err := GetAnsibleExecution(context.TODO(), helper, deployment, labels)
```

Comment:
Need to understand whether we actually want to re-use this Job, or whether we just check if it had succeeded. If it HAS succeeded, then we exit early on line 71. All other cases, we will be creating a new job. So I don't think we need to re-use this object in failure cases.